tools/clang/unittests/HLSL/ExtensionTest.cpp

```diff
@@ -97,6 +97,27 @@ static const HLSL_INTRINSIC_ARGUMENT TestMyBufferOp[] = {
   { "addr", AR_QUAL_IN, 1, LITEMPLATE_VECTOR, 1, LICOMPTYPE_UINT, 1, 2},
 };
 
+// float2 = CustomLoadOp(uint2 addr)
+static const HLSL_INTRINSIC_ARGUMENT TestCustomLoadOp[] = {
+  { "CustomLoadOp", AR_QUAL_OUT, 0, LITEMPLATE_VECTOR, 0, LICOMPTYPE_FLOAT, 1, 2 },
+  { "addr", AR_QUAL_IN, 1, LITEMPLATE_VECTOR, 1, LICOMPTYPE_UINT, 1, 2},
+};
+
+// float2 = CustomLoadOp(uint2 addr, bool val)
+static const HLSL_INTRINSIC_ARGUMENT TestCustomLoadOpBool[] = {
+  { "CustomLoadOp", AR_QUAL_OUT, 0, LITEMPLATE_VECTOR, 0, LICOMPTYPE_FLOAT, 1, 2 },
+  { "addr", AR_QUAL_IN, 1, LITEMPLATE_VECTOR, 1, LICOMPTYPE_UINT, 1, 2},
+  { "val",  AR_QUAL_IN, 2, LITEMPLATE_SCALAR, 2, LICOMPTYPE_BOOL, 1, 1},
+};
+
+// float2 = CustomLoadOp(uint2 addr, bool val, uint lo, uint hi)
```

Comment:
Should be float2 = CustomLoadOp(uint2 addr, bool val, uint2 subscript) ?

Reply:
Yep! Thanks, that's the DXIL version, not HLSL.